SCAutolib/cli_commands.py

```diff
@@ -165,7 +201,102 @@ def cleanup(ctx):
     exit(ReturnCode.SUCCESS.value)
 
 
-cli.add_command(setup_ca)
-cli.add_command(prepare)
-cli.add_command(setup_user)
-cli.add_command(cleanup)
+@cli.group(cls=NaturalOrderGroup, chain=True)
+@click.pass_context
+def gui(ctx):
+    """ Run GUI Test commands """
+    pass
+
+
+@gui.command()
+def init():
+    """ Initialize GUI for testing """
+    return "init"
+
+
+@gui.command()
+@click.argument("name")
+def assert_text(name):
+    """ Check if a word is found on the screen """
+    return f"assert_text:{name}"
+
+
+@gui.command()
+@click.argument("name")
+def assert_no_text(name):
+    """ Check that a word is not found on the screen """
+    return f"assert_no_text:{name}"
+
+
+@gui.command()
+@click.argument("name")
+def click_on(name):
+    """ Click on object containing word """
+    return f"click_on:{name}"
+
+
+@gui.command()
+def check_home_screen():
+    """ Check if screen appears to be the home screen """
+    return f"check_home_screen"
+
+
+@gui.command()
+@click.argument("keys")
+def kb_send(keys):
+    """ Send key(s) to keyboard """
+    return f"kb_send:{keys}"
+
+
+@gui.command()
+@click.argument("keys")
+def kb_write(keys):
+    """ Send string to keyboard """
+    return f"kb_write:{keys}"
+
+
+@gui.command()
+def done():
+    """ cleanup after testing """
+    return "done"
+
+
+@gui.result_callback()
+@click.pass_context
+def run_all(ctx, actions):
+    click.echo(actions)
+
+    from SCAutolib.models.gui import GUI
+    gui = GUI()
+    for action in actions:
+        if "init" in action:
+            gui.__enter__()
+        if "assert_text" in action:
+            assert_text = action.split(":")[1]
+            click.echo(f"CLICK: assert_text:{assert_text}")
+            gui.assert_text(assert_text)
+        if "assert_no_text" in action:
+            assert_text = action.split(":")[1]
+            click.echo(f"CLICK: assert_text:{assert_text}")
+            gui.assert_text(assert_text)
+        if "click_on" in action:
+            click_on = action.split(":")[1]
+            click.echo(f"CLICK: click_on:{click_on}")
+            gui.click_on(click_on)
+        if "check_home_screen" in action:
+            click.echo("CLICK: check_home_screen")
+            gui.check_home_screen()
+        if "kb_send" in action:
+            params = action.split(":")[1].split()
+            click.echo(f"CLICK: kb_send:{params}")
+            gui.kb_send(*params)
+        if "kb_write" in action:
```

Comment:
Maybe there is a good point to have the send command if you want to send some special character (somehow) but I guess that on kb_write it will be useful after we send something to press enter. What do you think?

Reply:
I have been thinking about that.   I added it to the cli_commands instead of directly in the GUI model so it wouldn't break anything existing.   Does that work?